src/android/ConnectPlugin.java

```diff
@@ -725,6 +735,67 @@ private void executeLogEvent(JSONArray args, CallbackContext callbackContext) th
         }
     }
 
+    private void executeSetUserID(JSONArray args, CallbackContext callbackContext) throws JSONException {
+      if (args.length() != 1) {
+          // Not enough parameters
+          callbackContext.error("Invalid arguments");
+          return;
+      }
+      String userID = args.getString(0);
+      logger.setUserID(userID);
+      callbackContext.success();
+      return;
+    }
+    
+    private void executeUpdateUserProperties(JSONArray args, final CallbackContext callbackContext) throws JSONException {
+      if (args.length() != 1) {
+          // Not enough parameters
+          callbackContext.error("Invalid arguments");
+          return;
+      }
+
+      JSONObject params = args.getJSONObject(0);
+      Bundle parameters = new Bundle();
+      Iterator<String> iter = params.keys();
+
+      while (iter.hasNext()) {
+          String key = iter.next();
+          try {
+              // Try get a String
+              String value = params.getString(key);
+              parameters.putString(key, value);
+              Log.w(TAG, "Key = " + key + " Value = " + value);
+          } catch (JSONException e) {
+              // Maybe it was an int
+              Log.w(TAG, "Type in AppEvent parameters was not String for key: " + key);
+              try {
+                  int value = params.getInt(key);
+                  parameters.putInt(key, value);
+              } catch (JSONException e2) {
+                  callbackContext.error("Unsupported type in AppEvent parameters for key: " + key);
+              }
+          }
+      }
+
+      logger.updateUserProperties(parameters, new GraphRequest.Callback() {
+          @Override
+          public void onCompleted(GraphResponse graphResponse) {
+              JSONObject jsonObj = graphResponse.getJSONObject();
+              try{
```

Comment:
I think you want spaces before/after brackets, based on how the rest of this file is coded